cli/azd/pkg/project/framework_service_swa.go

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package project
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/async"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/exec"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/swa"
+)
+
+type swaProject struct {
+	env           *environment.Environment
+	console       input.Console
+	commandRunner exec.CommandRunner
+	swa           swa.SwaCli
+	framework     FrameworkService
+}
+
+// NewSwaProject creates a new instance of a Azd project that
+// leverages swa cli for building
+func NewSwaProject(
+	env *environment.Environment,
+	console input.Console,
+	commandRunner exec.CommandRunner,
+	swa swa.SwaCli,
+	framework FrameworkService,
+) CompositeFrameworkService {
+	return &swaProject{
+		env:           env,
+		console:       console,
+		commandRunner: commandRunner,
+		swa:           swa,
+		framework:     framework,
+	}
+}
+
+// NewSwaProjectAsFrameworkService is the same as NewSwaProject().(FrameworkService) and exists to support our
+// use of DI and ServiceLocators, where we sometimes need to resolve this type as a FrameworkService instance instead
+// of a CompositeFrameworkService as [NewSwaProject] does.
+func NewSwaProjectAsFrameworkService(
+	env *environment.Environment,
+	console input.Console,
+	commandRunner exec.CommandRunner,
+	swa swa.SwaCli,
+	framework FrameworkService,
+) FrameworkService {
+	return NewSwaProject(env, console, commandRunner, swa, framework)
+}
+
+func (p *swaProject) Requirements() FrameworkRequirements {
+	return FrameworkRequirements{
+		Package: FrameworkPackageRequirements{
+			RequireRestore: false,
```

Comment:
Wondering if we need to take the inner framework's requirements into account here as well, so we we do things like `npm` restore if needed?

Reply:
All the requirements depend only and directly on `swa-config.json`.  Similar on how, for the Docker framework service, everything depends on the dockerfile.

When your service is using `swa-config.json`, `azd restore` should become a no-op because swa-cli does not support that operation:

![image](https://github.com/Azure/azure-dev/assets/24213737/00ac95ae-e25e-470e-8be4-9c17cde832d4)